app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -1315,8 +1309,38 @@ else if (CommCareApplication._().getAppResourceState() != CommCareApplication.ST
                 //Normal Home Screen login time!
                 refreshView();
             }
-        } catch (SessionUnavailableException sue) {
-            //TODO: See how much context we have, and go login
+        } catch(SessionUnavailableException sue) {
+            returnToInitialScreen();
+        }
+    }
+
+    /**
+     * Called in the event that we are leaving the CCHomeActivity (generally due to there no
+     * longer being an active session). Makes the decision of which prior screen or application
+     * state to return to, depending on the state of all currently installed CC apps.
+     */
+    private void returnToInitialScreen() {
+        // 1) If there is exactly one visible app and it's missing its MM, redirect to MM
+        // verification (because we're assuming the user is not using multiple apps)
+        if (CommCareApplication._().getVisibleAppRecords().size() == 1 && 
```

Comment:
I'm not really sure what change you're trying to indicate here. If there's something in the overall flow of this that you think should change, let's discuss verbally, because there are a lot of pieces to it.

As far as the specific relationship between visible/ready and missing multimedia, let's also talk about this verbally. I started writing out an explanation here and then realized I was essentially repeating the comments I have in the code, so I'm clearly not conveying it as clearly as I think I am.